src/ui/public/courier/__tests__/saved_object.js

```diff
@@ -0,0 +1,213 @@
+
+      let savedObject = createSavedObject(config);
+      expect(!!savedObject.searchSource.get('index')).to.be(false);
+
+      return savedObject.init().then(() => {
+        expect(afterESRespCallback.called).to.be(true);
+        const index = savedObject.searchSource.get('index');
+        expect(index instanceof IndexPattern).to.be(true);
+        expect(index.id).to.equal(indexPatternId);
+      });
+    });
+
+    describe('type', function () {
+      it('that is not specified throws an error', function () {
+        let config = {};
+
+        let savedObject = createSavedObject(config);
+        try {
+          savedObject.init();
+          expect(false).to.be(true);
+        } catch (err) {
+          expect(err).to.not.be(null);
+        }
+      });
+
+      it('that is invalid invalid throws an error', function () {
+        let config = {type: 'notypeexists'};
+
+        let savedObject = createSavedObject(config);
+        try {
+          savedObject.init();
+          expect(false).to.be(true);
+        } catch (err) {
+          expect(err).to.not.be(null);
+        }
+      });
+
+      it('that is valid passes', function () {
+        let config = {type: 'dashboard'};
+        return createSavedObject(config).init();
+      });
+    });
+
+    describe('defaults', function () {
+      it('applied to object with no id', function () {
+        let config = {
+          defaults: {testDefault: 'hi'},
+          type: 'dashboard'
+        };
+
+        let savedObject = createSavedObject(config);
+        return savedObject.init().then(() => {
+          expect(savedObject.searchSource).to.be(undefined);
+          expect(savedObject.defaults.testDefault).to.be(config.defaults.testDefault);
+        });
+      });
+
+      it('applied to source if an id is given', function () {
+        const myId = 'myid';
+        const customDefault = 'hi';
+        const initialOverwriteMeValue = 'this should get overwritten by the server response';
+
+        let config = {
+          defaults: {
+            overwriteMe: initialOverwriteMeValue,
+            customDefault: customDefault
+          },
+          type: 'dashboard',
+          id: myId
+        };
+
+        const serverValue = 'this should override the initial default value given';
+
+        const mockDocResponse = {
+          _source: {
+            overwriteMe: serverValue
+          },
+          _index: myId,
+          _type: 'dashboard',
+          _id: myId,
+          found: true
+        };
+
+        stubESResponse(mockDocResponse);
+
+        let savedObject = createSavedObject(config);
+        return savedObject.init().then(() => {
+          expect(!!savedObject._source).to.be(true);
+          expect(savedObject.defaults.overwriteMe).to.be(initialOverwriteMeValue);
+          expect(savedObject._source.overwriteMe).to.be(serverValue);
+          expect(savedObject._source.customDefault).to.be(customDefault);
+        });
+      });
+    });
+  });
+});
```

Comment:
Added one for applyESResp.  Save I'd prefer to get in the save/rename flow since I know what to test there.  As for the others, it feels like I'd just be testing implementation details as opposed to expected results, so at least with my current understanding of this code, it doesn't seem super beneficial.  (e.g. for delete, I could test that es.delete is called, but not that the item is actually deleted).